✨ [#11] 회원 정보 조회 기능 추가

ramleeramlee opened the pull request:
- ## #️⃣연관된 이슈
- 이슈 번호: #11

## 작업 내용
- 이메일, 캐시, 포인트, isAdmin 정보를 조회할 수 있습니다

## 체크리스트
- [x] 코드가 제대로 동작하는지 확인했습니다.
- [x] 관련 테스트를 추가했습니다.

ramleeramlee committed:
✨ [#11] 회원 정보 조회 기능 추가

- 이메일, 캐시, 포인트, isAdmin 정보를 조회할 수 있습니다

je-pa commented on goodsending/src/main/java/com/goodsending/member/service/MemberService.java:
```diff
@@ -42,5 +52,11 @@ public ResponseEntity<?> signup(SignupRequestDto signupRequestDto) {
     memberRepository.save(member);
     return ResponseEntity.ok("가입 완료");
+  // 회원 정보 조회
+  public MemberInfoDto getMemberInfo(Long memberId) {
```
MemberInfoDto 파일이 해당 브런치에 올라오지 않은 것 같아요~

ramleeramlee replied:
수정했습니다.

je-pa commented on goodsending/src/main/java/com/goodsending/member/service/MemberService.java:
```diff
@@ -16,6 +24,8 @@ public class MemberService {
 
   private final MemberRepository memberRepository;
   private final PasswordEncoder passwordEncoder;
+  private final MailService mailService;
```
MailServie import에 없으셔서 오류날것같아요~

ramleeramlee replied:
확인했습니다.
MemberService는 아직 수정중인 파일인데 나중에 올려도 되나요??

je-pa replied:
해당 코드 부분 삭제해주신 것 확인했습니다~

je-pa commented on goodsending/src/main/java/com/goodsending/member/service/MemberService.java:
```diff
@@ -42,5 +52,11 @@ public ResponseEntity<?> signup(SignupRequestDto signupRequestDto) {
     return ResponseEntity.ok("가입 완료");
+  // 회원 정보 조회
+  public MemberInfoDto getMemberInfo(Long memberId) {
+    MemberDetailsDto memberDetailsDto = memberRepository.findByMemberId(memberId);
```
예외 처리 해주시면 좋을 것 같아요~
이 메서드도 11번 브런치에 안보이는 것 같아요~

je-pa reviewed (commented):
고생하셨습니다.

ramleeramlee committed:
✨ [#11] 빠트린 파일 추가

ramleeramlee committed:
Merge branch 'dev' of https://github.com/goods-ending/goodsending-be into feature/#11

ramleeramlee committed:
🐛 [#11] bug 수정

- 브랜치끼리 충돌 되는 부분 수정
- Member 엔티티 cash, point 타입 Integer 변경
- dto 폴더 request/response 으로 세분화 및 위치 이동
- MemberInfoDto true/false가 아닌 role 값으로 나오도록 수정

je-pa commented on goodsending/src/main/java/com/goodsending/member/dto/response/MemberInfoDto.java:
```diff
@@ -0,0 +1,24 @@
+@Getter
+@NoArgsConstructor
+public class MemberInfoDto {
+  private Long member_id;
```
파스칼 케이스로 해주세요~

ramleeramlee replied:
수정했습니다.

je-pa commented on goodsending/src/main/java/com/goodsending/member/service/MemberService.java:
```diff
@@ -43,4 +45,9 @@ public ResponseEntity<?> signup(SignupRequestDto signupRequestDto) {
   }
+  // 회원 정보 조회
+  public MemberInfoDto getMemberInfo(Long memberId) {
+    MemberDetailsDto memberDetailsDto = memberRepository.findByMemberId(memberId);
```
바로 MemberDetailsDto로 받기 보다는 Optional로 받으셔서 null 체크 해주시는게 좋을 것 같아요~
여기서 Member엔티티가 아닌 MemberDetailsDto를 사용하신 이유가 있으실까요?

ramleeramlee replied:
저번에 Member 엔티티 사용안하는게 좋다고 하셔가지구 member 대신 사용했는데 Member 엔티티 그대로 써도 되나요?

je-pa replied:
그건 UserDetails 객체가 인증 및 권한에 사용되기 때문에 보안적으로 좋지 않기 때문이였어요!
service에서 Member 호출하고 Dto로 바로 변환해주는 것은 문제 없을 것 같아요!

ramleeramlee replied:
네 ㅎㅎ 이해했습니당

je-pa reviewed (commented):
고생하셨습니다~

ramleeramlee committed:
♻️ [#11] 코드 수정

- Optional으로 변경
- MemberDetailsDto 대신 Member 엔티티 사용으로 변경
- 예외처리 추가

je-pa commented on goodsending/src/main/java/com/goodsending/member/repository/MemberRepository.java:
```diff
@@ -1,5 +1,6 @@
 package com.goodsending.member.repository;
 
+import com.goodsending.member.dto.response.MemberDetailsDto;
```
import 문 제거해주시면 될 것 같아요!

ramleeramlee replied:
수정했습니다.

je-pa commented on goodsending/src/main/java/com/goodsending/member/dto/response/MemberDetailsDto.java:
```diff
@@ -11,14 +11,19 @@ public class MemberDetailsDto {
   private final Long memberId;
   private final String email;
   private final String password;
+  private final Integer cash;
+  private final Integer point;
   private final MemberRole role;
 
   public static MemberDetailsDto from(Member member) {
     return new MemberDetailsDto(
         member.getMemberId(),
         member.getEmail(),
         member.getPassword(),
+        member.getCash(),
+        member.getPoint(),
         member.getRole()
     );
```
다시 보니 그때 security 인증부분에 사용하려고 분리했던 Dto 였군요!
보안적인 부분과 관점 분리의 목적으로 분리했던 Dto여서 인증관련 부분만 적어주시면 좋겠습니다!

ramleeramlee replied:
네!! 수정했습니다.

je-pa reviewed (commented):
고생하셨습니다~

ramleeramlee committed:
♻️ [#11] 코드 수정

- 사용하지 않는 import 제거
- MemberDetailsDto 인증시 필요 없는 부분 일부 삭제
